src/memray/_memray/sink.cpp

```diff
@@ -181,6 +184,20 @@ FileSink::~FileSink()
     if (d_fd != -1) {
         ::close(d_fd);
     }
+
+    if (d_compress) {
+        std::ifstream in_file(d_filename);
+        std::string tmp_filename = d_filename + ".lz4";
+        std::ofstream out_file(tmp_filename);
+        lz4_stream::ostream lz4_stream(out_file);
+        std::copy(
+                std::istreambuf_iterator<char>(in_file),
+                std::istreambuf_iterator<char>(),
+                std::ostreambuf_iterator<char>(lz4_stream));
```

Comment:
Surely it's inefficient to copy one character at a time...

Reply:
Haha, sure, but I will use the "this was a draft so get me out of the jail card" 😉